blog/utils/handlePosts.ts

```diff
@@ -48,6 +47,21 @@ export const filterPostsByCategory = (posts: BlogPost[], slug?: string) =>
     ? posts.filter(({ categories }) => categories.find((c) => c.slug === slug))
     : posts;
 
+/**
+ * Returns an filtered BlogPost list
+ *
+ * @param posts Posts to be handled
+ * @param term Term to be filter
+ */
+export const filterPostsByTerm = (posts: BlogPost[], term?: string) =>
+  term
+    ? posts.filter(({ content, excerpt, title }) =>
+      [content, excerpt, title].some((field) =>
+        field.toLowerCase().includes(term.toLowerCase())
+      )
+    )
+    : posts;
+
```

Comment:
```suggestion
export const filterPostsByTerm = (posts: BlogPost[], term: string) =>

    posts.filter(({ content, excerpt, title }) =>
      [content, excerpt, title].some((field) =>
        field.toLowerCase().includes(term.toLowerCase())
      )
    );

```

Reply:
you verify the existence of term before call this function, i think you could do the same in  `filterPostsByCategory`